src/decorators/registry.ts

```diff
@@ -0,0 +1,25 @@
+import { handleDecorator } from './handleDecorator';
+import { RegistryItem } from '../Registry';
+
+export interface RegistryConfig {
+	[ name: string ]: RegistryItem;
+}
+
+/**
+ * Decorator that can be used to register a widget with the registry
```

Comment:
Perhaps explain that it will register the widget with the widget's local registry?